java/analyzer/src/main/java/org/wikidata/analyzer/Processor/ReferenceProcessor.java

```diff
@@ -0,0 +1,417 @@
+package main.java.org.wikidata.analyzer.Processor;
+
+import com.google.common.collect.Iterators;
+import org.wikidata.wdtk.datamodel.interfaces.*;
+
+import java.util.*;
+
+/**
+ * @author Addshore
+ */
+public class ReferenceProcessor implements EntityDocumentProcessor {
+
+    private Map<String, Long> counters;
+
+    private Map<String, String> wikimedias = new HashMap<>();
+
+    private List<String> referenceProperties = new ArrayList<>();
+
+    public ReferenceProcessor(Map<String, Long> counters) {
+        this.counters = counters;
+        this.populateWikimedias();
+        this.populateReferenceProperties();
+    }
+
+    /**
+     * The below list was generated using a SPARQL query.
+     * FIXME: there is probably a better way to store, manage and update this....
+     */
+    private void populateWikimedias() {
```

Comment:
Yeh! :)
Also, woo tests!
I'll have to spend some time working out how to make travis run the java tests with maven at some point (unless you want to) ;)